lib/membrane_mp4/demuxer/isom.ex

```diff
@@ -460,15 +460,16 @@ defmodule Membrane.MP4.Demuxer.ISOM do
     Range.size(tracks) == length(pads)
   end
 
-  defp flush_samples(state, track_id) do
-    buffers =
-      Map.get(state.buffered_samples, track_id, [])
-      |> Enum.reverse()
-      |> Enum.map(fn {buffer, ^track_id} -> buffer end)
+  defp flush_samples(state) do
+    Enum.flat_map_reduce(state.buffered_samples, state, fn {track_id, track_samples}, state ->
+      buffers =
+        track_samples
+        |> Enum.reverse()
+        |> Enum.map(fn {buffer, ^track_id} -> buffer end)
 
-    actions = [buffer: {Pad.ref(:output, track_id), buffers}]
-
-    {actions, put_in(state, [:buffered_samples, track_id], [])}
+      new_actions = [buffer: {Pad.ref(:output, track_id), buffers}]
+      {new_actions, put_in(state, [:buffered_samples, track_id], [])}
```

Comment:
NIT: I'd reset buffered samples in another run, so this actually could become a flat_map

Reply:
I have used `map`